felix/nftables/ipsets.go

```diff
@@ -693,7 +722,7 @@ func (s *IPSets) ApplyDeletions() bool {
 
 	// ApplyDeletions() marks the end of the two-phase "apply". Piggyback on that to
 	// update the gauge that records how many IP sets we own.
-	numDeletionsPending := s.setNameToProgrammedMetadata.Dataplane().Len()
+	numDeletionsPending := s.setNameToProgrammedMetadata.PendingDeletions().Len()
```

Comment:
@fasaxc I noticed this line seems wrong, and that I copied it straight from the ipset implementation: https://github.com/projectcalico/calico/blob/ade2595a23c5b44cea5cea012545b1ef67eb7cee/felix/ipsets/ipsets.go#L963-L964

Is it actually just wrong there as well?

Reply:
I suppose it's a bit confusing, because it's called `numDeletionsPending` but it's setting the `gaugeNumSets` metric, and being used to return a value indicating whether or not additional deletions need to be made.

So I think we want to switch this to `PendingDeletions()` and then also update the gauge from `Dataplane().Len()` separately.

Reply:
Yep, your version looks right, no idea what I did there.